File: validation/run_validation_all.R
# ===================================================================
# Run All Validation Stages (R Orchestrator)
#
# Executes available stages in sequence.
# ===================================================================

source("validation/common_config.R")
source("validation/stage_01_robust_stats.R")
source("validation/stage_02_homogeneity.R")
source("validation/stage_03_stability.R")
source("validation/stage_04_uncertainty_chain.R")
source("validation/stage_05_scores.R")

run_validation_all <- function() {
  combos <- get_target_combos()
  validate_combo_definition(combos)

  cat("Validation pipeline started.\n")
  cat(sprintf("Target combos loaded: %d\n", nrow(combos)))
  cat("Running stage_01_robust_stats...\n")
  run_stage_01_robust_stats()
  cat("Running stage_02_homogeneity...\n")
  run_stage_02_homogeneity()
  cat("Running stage_03_stability...\n")
  run_stage_03_stability()
  cat("Running stage_04_uncertainty_chain...\n")
  run_stage_04_uncertainty_chain()
  cat("Running stage_05_scores...\n")
  run_stage_05_scores()
  cat("Validation pipeline finished.\n")
}

if (identical(environment(), globalenv())) {
  run_validation_all()
}

File: validation_2/run_validation_all.py
"""Run all validation stages (Python orchestrator).

Executes available stages in sequence.
"""

from common_config import get_target_combos, validate_combo_definition
from stage_01_robust_stats import run_stage_01_robust_stats
from stage_02_homogeneity import run_stage_02_homogeneity
from stage_03_stability import run_stage_03_stability
from stage_04_uncertainty_chain import run_stage_04_uncertainty_chain
from stage_05_scores import run_stage_05_scores


def run_validation_all() -> None:
  combos = get_target_combos()
  validate_combo_definition(combos)

  print("Validation pipeline started.")
  print(f"Target combos loaded: {len(combos)}")
  print("Running stage_01_robust_stats...")
  run_stage_01_robust_stats()
  print("Running stage_02_homogeneity...")
  run_stage_02_homogeneity()
  print("Running stage_03_stability...")
  run_stage_03_stability()
  print("Running stage_04_uncertainty_chain...")
  run_stage_04_uncertainty_chain()
  print("Running stage_05_scores...")
  run_stage_05_scores()
  print("Validation pipeline finished.")


if __name__ == "__main__":
  run_validation_all()
